src/balancerd/tests/server.rs

```diff
@@ -127,6 +126,8 @@ fn test_balancer() {
         None,
     )
     .unwrap();
+    task::spawn(|| "test_balancer-frontegg_future", frontegg_future);
```

Comment:
This is pretty ugly. After both these PRs merge we should do a similar refactor to the frontegg server so it's async-first also.

Reply:
Totally agree, once you merge https://github.com/MaterializeInc/materialize/pull/23176 I'll rebase and have a think about how to make this cleaner

Reply:
After rebasing on top of https://github.com/MaterializeInc/materialize/pull/23176 I moved the `task::spawn` into the `FronteggMockServer` itself, so we no longer need these `spawn` calls!